src/app/workspace/resource/resource-instance-form/resource-instance-form.component.ts

```diff
@@ -350,66 +335,4 @@ export class ResourceInstanceFormComponent implements OnInit, OnDestroy {
 
     }
 
-    /**
-     * Given a map of class definitions,
-     * returns an array of resource class definitions.
-     *
-     * @param classDefs a map of class definitions
-     */
-    private _makeResourceClassesArray(classDefs: { [index: string]: ClassDefinition }): ResourceClassDefinition[] {
-
-        const classIris = Object.keys(classDefs);
-
-        // get resource class defs
-        return classIris.filter(resClassIri => {
-            return typeGuard(classDefs[resClassIri], ResourceClassDefinition);
-        }).map(
-            (resClassIri: string) => {
-                return classDefs[resClassIri] as ResourceClassDefinition;
-            }
-        );
-
-    }
-
-    /**
-     * Given a map of property definitions,
-     * returns a map of resource property definitions.
-     *
-     * @param propertyDefs a map of property definitions
-     */
-    private _makeResourceProperties(propertyDefs: { [index: string]: PropertyDefinition }): Properties {
-        const resProps: Properties = {};
-
-        const propIris = Object.keys(propertyDefs);
-
-        propIris.filter(
-            (propIri: string) => {
-                return typeGuard(propertyDefs[propIri], ResourcePropertyDefinition);
-            }
-        ).forEach((propIri: string) => {
-            resProps[propIri] = (propertyDefs[propIri] as ResourcePropertyDefinition);
-        });
-
-        return resProps;
-    }
-
-    private convertPropObjectAsArray() {
-        // represent the properties as an array to be accessed by the template
-        const propsArray = [];
-
-        for (const propIri in this.properties) {
-            if (this.properties.hasOwnProperty(propIri)) {
-                const prop = this.properties[propIri];
-
-                // only list editable props that are not link value props
-                if (prop.isEditable && !prop.isLinkProperty) {
```

Comment:
I could reproduce the problem locally. I think you need to put this logic in `ResourceInstanceFormComponent.selectProperties`:

```ts
    selectProperties(resourceClassIri: string) {

        ...

        // filter out all props that cannot be edited or are link props 
        this.properties = onto.getPropertyDefinitionsByType(ResourcePropertyDefinition).filter(prop => prop.isEditable && !prop.isLinkProperty);
        
          
        ....
```

Reply:
Otherwise the form creates entries for system props like `knora-api:hasIncomingLinkValue` which you cannot edit.
In the case at hand, the problem was that `knora-api:hasIncomingLinkValue` had no corresponding def. for the link prop. (`knora-api:hasIncomingLink`)